src/System.Collections/tests/Generic/HashSet/HashSet.Generic.Tests.cs

```diff
@@ -162,6 +162,122 @@ public void HashSet_Generic_RemoveWhere_NullMatchPredicate(int setLength)
 
         #endregion
 
+        #region EnsureCapacity
+        
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void EnsureCapacity_Generic_RequestingLargerCapacity_DoesNotInvalidateEnumeration(int setLength)
+        {
+            HashSet<T> set = (HashSet<T>)(GenericISetFactory(setLength));
+            var capacity = set.EnsureCapacity(0);
+            IEnumerator<T> valuesEnum = set.GetEnumerator();
+            set.EnsureCapacity(capacity + 1); // Verify EnsureCapacity does not invalidate enumeration
+            while (valuesEnum.MoveNext());
```

Comment:
In addition to validating that it doesn't throw, it also be good to validate that you still get the data you expect.